src/Security/MemberAuthenticator/ChangePasswordHandler.php

```diff
@@ -348,4 +388,48 @@ protected function checkPassword($member, $password)
         }
         return true;
     }
+
+    /**
+     * Generate a temporary auto login hash for the member
+     */
+    private function createAutoLoginTempHash(): string
+    {
+        if (is_callable($this->autoLoginTempHashGenerator)) {
+            return call_user_func($this->autoLoginTempHashGenerator);
+        }
+        return $this->createDefaultAutoLoginTempHash();
+    }
+
+    /**
+     * Generate a temporary auto login hash for the member
+     */
+    private function createDefaultAutoLoginTempHash(): string
+    {
+        $member = null;
+        $autoLoginTempHash = '';
+        do {
+            $autoLoginTempHash = Injector::inst()->get(RandomGenerator::class)->randomToken('sha256');
```

Comment:
The benefit of using injector here is so that we don't need `createAutoLoginTempHash()` calling this method (just have the one method), and we don't need the `autoLoginTempHashGenerator` property nor its setter.

Instead, just set a dummy token generator into the injector for the test as suggested in https://github.com/silverstripe/silverstripe-framework/pull/11668#discussion_r2032095427